packages/events-example/src/commands/reconcile/index.ts

```diff
@@ -1,84 +1,117 @@
   nile!: NileApi;
 
-  async run(): Promise<any> {
+  /**
+   * Runner for oclif
+   * @returns void
+   */
+  async run(): Promise<unknown> {
     const { flags } = await this.parse(Reconcile);
 
-    // nile setup
-    await this.connectNile(flags);
-
-    // pulumi setup
-    this.deployment = await PulumiAwsDeployment.create("nile-examples", pulumiProgram, { region: flags.region });
-
-    // load our data
-    const stacks = await this.deployment.loadPulumiStacks();
-    const instances = await this.loadNileInstances(
-      flags.organization,
-      flags.entity
+    const {
+      status,
+      region,
+      organization,
+      entity,
+      basePath,
+      workspace,
+      email,
+      password,
+    } = flags;
+
+    // Nile setup
+    await this.connectNile({ basePath, workspace, email, password });
+    const instances = await this.loadNileInstances(organization, entity);
+
+    // Pulumi setup
+    this.deployment = await PulumiAwsDeployment.create(
+      'nile-examples',
+      pulumiS3,
+      { region }
     );
+    const stacks = await this.deployment.loadPulumiStacks();
+
+    // stitch Nile and Pulumi together
     const plan = new ReconciliationPlan(instances, stacks);
 
-    if (flags.status) {
+    if (status) {
       this.log('Status check only.');
-      this.log(`Pending destruction: ${ plan.destructionIds } (${ plan.destructionIds.length})`);
-      this.log(`Pending creation: ${ plan.creationIds } (${ plan.creationIds.length })`)
+      this.log(
+        `Pending destruction: ${plan.destructionIds} (${plan.destructionIds.length})`
+      );
+      this.log(
+        `Pending creation: ${plan.creationIds} (${plan.creationIds.length})`
+      );
       return { stacks, instances };
     }
 
+    // load or remove stacks based on Nile
     await this.synchronizeDataPlane(plan);
-    await this.listenForNileEvents(flags.entity, this.findLastSeq(Object.values(instances)));
+
+    // listen to updates from nile and handle stacks accordingly
+    await this.listenForNileEvents(
+      flags.entity,
+      this.findLastSeq(Object.values(instances))
+    );
   }
 
-  async connectNile(parsedFlags: {[name: string]: any}) {
-    this.nile = Nile({ basePath: parsedFlags.basePath, workspace: parsedFlags.workspace });
+  /**
+   * sets up Nile instance, and set auth token to the logged in developer
+   * @param config Configuration for instantiating Nile and logging ing
+   */
+  async connectNile({
+    basePath,
+    workspace,
+    email,
+    password,
+  }: NileConfig & DeveloperCreds) {
+    const developerPayload = {
+      loginInfo: {
+        email,
+        password,
+      },
+    };
+    this.nile = Nile({
+      basePath,
+      workspace,
+    });
     const token = await this.nile.developers
-      .loginDeveloper({
-        loginInfo: {
-          email: parsedFlags.email,
-          password: parsedFlags.password,
-        },
-      })
+      .loginDeveloper(developerPayload)
```

Comment:
Nit: you've got the Nile creds embedded in the constructor, but a separate `developerPayload` declared. I don't feel like we lose any clarity embedding them here vs. a separate declaration.

Reply:
indeed. I think this is an issue with the way oclif is doing things. I wanted it very clear in the code about what you need from nile vs every where else. In the case you don't want to use the CLI (I wouldn't, I think I want to convert this example an agent), splitting what oclif wants us to do vs what we want a developer to do is worth while